src/go/k8s/pkg/console/configmap.go

```diff
@@ -51,7 +53,18 @@ func NewConfigMap(
 
 // Ensure implements Resource interface
 func (cm *ConfigMap) Ensure(ctx context.Context) error {
-	if cm.consoleobj.Status.ConfigMapRef != nil {
+	if ref := cm.consoleobj.Status.ConfigMapRef; ref != nil { //nolint:nestif // need to check for each error
+		// Check ConfigMap is present, in case it is manually deleted
+		if err := cm.Get(ctx, client.ObjectKey{Namespace: ref.Namespace, Name: ref.Name}, &corev1.ConfigMap{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				cm.consoleobj.Status.ConfigMapRef = nil
+				if updateErr := cm.Status().Update(ctx, cm.consoleobj); updateErr != nil {
+					return updateErr
+				}
+				return &resources.RequeueError{Msg: err.Error()}
+			}
+			return err
+		}
 		return nil
 	}
```

Comment:
My proposal would be 
```suggestion
	if ref := cm.consoleobj.Status.ConfigMapRef; ref != nil {
		// Check ConfigMap is present, in case it is manually deleted
		err := cm.Get(ctx, client.ObjectKey{Namespace: ref.Namespace, Name: ref.Name}, &corev1.ConfigMap{})
		if apierrors.IsNotFound(err) {
			cm.consoleobj.Status.ConfigMapRef = nil
			if updateErr := cm.Status().Update(ctx, cm.consoleobj); updateErr != nil {
				return &resources.RequeueError{Msg: updateErr.Error()}
			}
			return &resources.RequeueError{Msg: err.Error()}
		}
		return err

	}
```

Reply:
So just remove the extra if statement? Updated.